app/controllers/collaborators_controller.rb

```diff
@@ -0,0 +1,27 @@
+class CollaboratorsController < ApplicationController
+   before_action :authenticate_user!
+ 
+   def create
+     wiki = Wiki.find(params[:wiki_id])
+     collaborator = current_user.collaborators.build(wiki: wiki)
+ 
+     if collaborator.save
+       flash[:notice] = "Collaborator created"
+     else
+       flash[:alert] = "Collaborator creation failed."
+     end
+     redirect_to wikis_path
+   end
+   
+   def destroy
+     wiki = Wiki.find(params[:id])
+     collaborator = current_user.collaborators.where(wiki: wiki).create
```

Comment:
Line 18:
 - no need for a `create` (we're in the `destroy` action)
 - `current_user.collaborators.where(wiki: wiki)` returns a collection of collaborators not a single collaborator

Consider changing line 18:
```diff
- collaborator = current_user.collaborators.where(wiki: wiki).create
+ collaborator = current_user.collaborators.find_by!(wiki: wiki)
```